_books/ion-1-1/src/modules/defining_modules.md

```diff
@@ -0,0 +1,242 @@
+
+symbol-list        ::= '[' ((symbol | string)',')* ']'
+```
+
+The `symbol_table` clause assembles a list of text values for the module to export.
+It takes any number of arguments, each of which may be the name of visible module or a list of symbol-texts.
+The symbol table is a list of symbol-texts by concatenating the symbol tables of named modules and lists of symbol/string values.
+
+Where a module name occurs, its symbol table is appended.
+(The module name must refer to another module that is visible to the current module.)
+Unlike Ion 1.0, no _symbol-maxid_ is needed because Ion 1.1 always required exact matches for imported modules.
+
+Where a list occurs, it must contain only non-null, unannotated strings and symbols.
+The text of these strings and/or symbols are appended to the symbol table.
+Upon encountering any non-text value, null value, or annotated value in the list, the implementation shall signal an error.  
+To add a symbol with unknown text to the symbol table, one may use `$0`.
+
+All modules have a symbol table, so when a module has no `symbol_table` clause, the module has an empty symbol table. 
+
+#### Processing
+
+When the `symbol_table` clause is encountered, the reader constructs an empty list. The arguments to the clause are then processed from left to right.
+
+For each `arg`:
+* **If the `arg` is a list of text values**, the nested text values are appended to the end of the symbol table being constructed.
+  * When `$0` appears in the list of text values, this creates a symbol with unknown text.
+  * The presence of any other Ion value in the list raises an error.
+* **If the `arg` is the name of a module**, the symbols in that module's symbol table are appended to the end of the symbol table being constructed.
+* **If the `arg` is anything else**, the reader must raise an error.
+
+#### Example
+
+```ion
+(symbol_table         // Constructs an empty symbol table (list)
+  ["a", b, 'c']       // The text values in this list are appended to the table
+  foo                 // Module `foo`'s symbol table values are appended to the table
+  ['''g''', "h", i])  // The text values in this list are appended to the table
+```
+If module `foo`'s symbol table were `[d, e, f]`, then the symbol table defined by the above clause would be:
+```ion
+["a", "b", "c", "d", "e", "f", "g", "h", "i"]
+```
+This is an Ion 1.0 symbol table that imports two shared symbol tables and then declares some symbols of its own.
+
+```ion
+$ion_1_0
+$ion_symbol_table::{
+  imports: [{ name: "com.example.shared1", version: 1, max_id: 10 },
+            { name: "com.example.shared2", version: 2, max_id: 20 }],
+  symbols: ["s1", "s2"]
+}
+```
+Here’s the Ion 1.1 equivalent in terms of symbol allocation order:
+
+```ion
+$ion_1_1
+$ion_encoding::(
+  (import m1 "com.example.shared1" 1)
+  (import m2 "com.example.shared2" 2)
+  (symbol_table m1 m2 ["s1", "s2"])
+)
+```
+
+
+
+### `macro_table`
+
+Macros are declared after symbols.
```

Comment:
Is there a specific reason for this ordering? Food for thought: if symbols were declared after macros, would that make it clearer that the symbols defined in an encoding directive are not yet available for use in the macro definitions in that same encoding directive?

Reply:
Created https://github.com/amazon-ion/ion-docs/issues/356